src/Microsoft.Framework.Runtime/Servicing/ServicingIndex.cs

```diff
@@ -83,22 +90,30 @@ public void Initialize(string folderPath)
 
         public bool TryGetReplacement(string packageId, SemanticVersion packageVersion, string assetPath, out string replacementPath)
         {
+            replacementPath = null;
+
             Entry entry;
             if (_entries.TryGetValue(new EntryKey(packageId, packageVersion), out entry))
             {
                 var normalizedAssetPath = assetPath.Replace('\\', '/');
+
+                // Go through the entire list of mappings and use the latest one
                 foreach (var mapping in entry.Mappings)
                 {
                     if (string.Equals(normalizedAssetPath, mapping.AssetPath, StringComparison.OrdinalIgnoreCase))
                     {
-                        replacementPath = Path.Combine(_folderPath, mapping.ReplacementPath);
```

Comment:
Just an idea. Will the entry mappings be big? If so, would it be faster if just reverse the sequence and return the first hit?